### ISSUE TITLE
Is an iPadOS version possible / viable?

### ISSUE THREAD

**Issue by vegabook** [wez/wezterm#6069]
There are no terminal emulators on ipadOS that support sixel graphics, that I've been able to find.

a) would it be difficult to compile from source for iOS? What are the dependencies that one might run into trouble with?

b) Have you considered building an iOS version?

I think once you have Sixel graphics a lot of data science (charting) can be done right in the terminal (I'm already doing this on mac and Linux), and I would love to use my (very conventient, small form factor) ipad on the road for this.

**Comment by wez** [comment]
It should be technically possible to port to iOS, but likely there would need to be a number of code changes for it to actually compile and run, so it is not something you can directly try today.

I don't think Apple will allow wezterm to ship to its app store with local execution domain support though, so you're probably looking at making an ssh-focused implementation of wezterm.

There are a lot of considerations that would need to be made for a tablet form factor before wezterm would be an actually good user experience there.

**Comment by vegabook** [comment]
yes it's the SSH-focused version which I thought would be interesting. The other terminals on iPad are much too high level and don't do basic things one expected but if one could use the iPad as a fully featured wezterm with which to ssh home, that would be awesome. But sounds like it's a tough gig but thanks for answering.

**Comment by sander-auguria** [comment]
Currently using blink on ipad, pretty happy with its functionality, able to keep app alive in background, port-forward over ssh and use mosh reliably into a multiplex like tmux or zellij.  

> I don't think Apple will allow wezterm to ship to its app store with local execution domain support though, so you're probably looking at making an ssh-focused implementation of wezterm.

I'm not sure that would work with this restriction though.

**Comment by nickvikeras** [comment]
> It should be technically possible to port to iOS, but likely there would need to be a number of code changes for it to actually compile and run, so it is not something you can directly try today.
> 
> 
> 
> I don't think Apple will allow wezterm to ship to its app store with local execution domain support though, so you're probably looking at making an ssh-focused implementation of wezterm.
> 
> 
> 
> There are a lot of considerations that would need to be made for a tablet form factor before wezterm would be an actually good user experience there.
> 
> 

Being able to use wezterm as an ssh mux client on an iPad would be so cool.

**Comment by 7heaven** [comment]
Secure Terminal does support sixel graphics, you can find it in ipad's app store
